lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -92,23 +92,25 @@ def set_location(self, location):
 
         # This puts the rectangle into figure-relative coordinates.
         if isinstance(location, str):
+            _api.check_in_list(self._locstrings, location=location)
             if location in ['top', 'right']:
                 self._pos = 1.
-            elif location in ['bottom', 'left']:
+            else:  # 'bottom', 'left'
                 self._pos = 0.
-            else:
-                raise ValueError(
-                    f"location must be {self._locstrings[0]!r}, "
-                    f"{self._locstrings[1]!r}, or a float, not {location!r}")
-        else:
+        elif isinstance(location, float):
```

Comment:
This excludes putting in, e.g., `1` since it's an `int`, not a `float`. Might want `numbers.Real` here, though I'm not sure we use it much.

Reply:
There is a mix of `Number` and `Real` in the code. I guess there should be very few `Number`, if any, as most code won't appreciate complex numbers?

(Also one case of both float and int.)